library_generation/utils/utilities.sh

```diff
@@ -157,7 +161,22 @@ download_generator_artifact() {
 download_protoc() {
```

Comment:
In general, I feel like we may not need any changes in `download_protoc`. Ideally this function can stay as it is and we can add logics before calling it, for example
```
if [[ "${protoc_version}" != "${DOCKER_PROTOC_VERSION}" ]]; then
  download_protoc "${protoc_version}" "${os_architecture}"
fi
```
But I think this requires us to reuse the proto location specified in Dockerfile.

Reply:
I left `download_protoc` _almost_ untouched and left the `DOCKER_PROTOC_VERSION` handling in `download_tools` (one level up in the call hierarchy).